probe/probe.go

```diff
@@ -150,8 +173,14 @@ func (p *Probe) publishLoop() {
 	for {
 		select {
 		case <-pubTick:
-			localReport := p.rpt.swap(report.MakeReport())
-			if err := rptPub.Publish(localReport); err != nil {
+			rpt := condense(report.MakeReport(), p.spiedReports)
+			if err := rptPub.Publish(rpt); err != nil {
+				log.Printf("publish: %v", err)
+			}
+
+		case rpt := <-p.shortcutReports:
+			rpt = condense(rpt, p.shortcutReports)
+			if err := rptPub.Publish(rpt); err != nil {
```

Comment:
Yeah I was hoping for a way to unify these a bit more, but I couldn't come up with a nice way.  I guess we could move the publish step into the condense function, and call it drainAndPublish?